scania-digital-design-system/tegel
# feat: angular wrappers

**Describe pull-request**  
Angular wrappers for the Tegel package, easing Angular integration in using Tegel components.

This PR:
 - Creates a monorepo structure (packages -> core/angular)
 - Updates the package.json scripts for the core package to conform to the new structure
 - Adds the angularOutputTarget which builds Angular Wrappers
 - Updated the gitignore to not include any stencil/angular generated files or folders
 - Updated the eslintignore to conform to the new structure
 - Updated the README for the new `packages/angular` folder

**Solving issue**  
Fixes: [CDEP-1945](https://tegel.atlassian.net/jira/software/projects/CDEP/boards/1?assignee=63ff6d6c314f5088138187f0&selectedIssue=CDEP-1945)

**How to test**  
- Pull down the branch and run `npm install.`
- cd into `packages/core`
- Run `npm install`
- Run `npm run build`
- cd into `packages/angular`
- Run `npm i`
- Run `npm run build`
- Run `npm pack` in `packages/angular`, which will create a .tgz file
- Move on to create a test Angular application: `ng new name-of-app`, move the .tgz file into the angular app and then run `npm i ./scania-tegel-angular-0.0.1.tgz`
- Import the TegelModule in the components where you want to use tegel components:


```
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { AppComponent } from './app.component';
import {
  TegelModule,
} from '@scania/tegel-angular';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    TegelModule,
  ],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}


```


- In order to get the styling add `@import url('@scania/tegel/dist/tegel/tegel.css')` to your `styles.css` file. 
- In the template for that component, add: `<tds-button text="Angular baby!"></tds-button>`
- Finally run `npm start` and hopefully you will see your button component.



[CDEP-1945]: https://tegel.atlassian.net/browse/CDEP-1945?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
